tests/aws/services/sns/test_sns.py

```diff
@@ -5124,6 +5125,7 @@ def test_dlq_external_http_endpoint(
             "$.topic-attrs.Attributes.Policy.Statement..Action",
         ]
     )
+    # Is this even executable against AWS?
```

Comment:
yes it is 😄 but it is quite a complex setup: you need to make your local http server reachable from the internet to be able to see what messages SNS sent you. See the `sns_create_http_endpoint` fixture. 

Ultimately, I think we could replace this manual setup with a lambda Function URL instead and forward those received HTTP calls via SQS and retrieve them, but we wouldn't be able to test as much as Lambda will modify the incoming payload so it's not as precise as this setup. 

But this test was validated, and I believe is important that it passes because this was reported by users and was directly implementing the feature. 

How does it still fail today? What errors do we get?

Reply:
It was an error when raw message delivery was disabled, and it was one header only that was wrong. I thought that before I start a deep dive as to why one specific header fails, I want to clarify if this is even relevant :smile: But it is, and it was a bug in the publisher, needed another of these "access pascal case or CamelCase for topic attributes" fixes we talked about.